test/sparsematrix_ops.jl

```diff
@@ -442,4 +442,15 @@ end
     @test one(A) isa SparseMatrixCSC{Int}
 end
 
+@testset "transpose! does not allocate" begin
+    function f()
+        A = sprandn(10, 10, 0.1)
+        X = deepcopy(A)
```

Comment:
```suggestion
        X = copy(A)
```

Reply:
No `copy` copies:
https://github.com/JuliaSparse/SparseArrays.jl/blob/7745517d8e6f22553af0732bb7b018ee6390c893/src/sparsematrix.jl#L422-L423
You should basically never use `deepcopy`.

Reply:
It seems weird that it's implemented for AbstractSparseMatrixCSC, but interesting.

`deepcopy` is avoided because of dynamic dispatch?

Reply:
> > i don't know why
> 
> See https://docs.julialang.org/en/v1/manual/performance-tips/#Be-aware-of-when-Julia-avoids-specializing

not sure i understand why. The allocations are from dynamic dispatch on the functions?